### DISCUSSION TITLE
데이터의 아이디를 만들 때는 데이터의 길이로 만들어서는 안됩니다.

### DISCUSSION THREAD

**Discussion by hannut91** [CodeSoom/spring-week1-assignment-1#47] [General]
## 문제

데이터를 만들 때 데이터의 길이를 이용해서 만든다면, 아이디가 겹치는 문제가 생깁니다.

## 예시

```
1
2
3
```

이러한 데이터가 있을 때 중간 값을 삭제합니다.

```
1
3
```

이 상황에서 새로운 값을 하나 추가합니다. 길이(2) + 1을 더한 값을 아이디로 할당하니 다음과 같은 결과가 나옵니다.

```
1
3
3
```

## 해결

따라서 아이디를 만들 때는 데이터의 길이를 가지고 만들어서는 안됩니다. 그렇다면 아이디는 어떻게 만드는 것이 좋을까요?

**Comment by wjy5446** [comment]
NEXT_ID 라는 글로벌 변수를 만들고, 데이터가 생성할 때마다 해당 변수를 부여하고, +1를 해주는 방법을 쓰면 어떨까요?

**Comment by jxmen** [comment]
데이터의 `마지막 ID`+1은 어떨까요? 
그렇다면 중간이나 마지막 값을 삭제한다고 해도 ID 중복은 피할 수 있지 않을까요?

**Reply by hannut91** [reply]
ID 중복을 피할 수 있을 것 같습니다 ㅎㅎ 벌써 해결책을 찾으셨군요. 해결책을 찾았는데, 이제 우리는 더 좋은 방법은 없을까? 더 효율적인 방법은 없을까? 생각해봐야 합니다
